CMakeLists.txt

```diff
@@ -4,350 +4,109 @@ if(COMMAND CMAKE_POLICY)
+  # Find tinyxml2.
+  ign_find_package(TINYXML2 REQUIRED)
+
+  ################################################
+  # Find urdfdom parser. Logic:
+  #
+  #  1. if USE_INTERNAL_URDF is unset, try to use system installation, fallback to internal copy
+  #  2. if USE_INTERNAL_URDF is set to True, use the internal copy
+  #  3. if USE_INTERNAL_URDF is set to False, force to search system installation, fail on error
+  if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
+    ign_find_package(IgnURDFDOM VERSION 1.0 QUIET)
+    if (NOT IgnURDFDOM_FOUND)
+      if (NOT DEFINED USE_INTERNAL_URDF)
+        # fallback to internal urdf
+        set(USE_INTERNAL_URDF ON)
+      else()
+        ign_build_error("Couldn't find the urdfdom >= 1.0 system installation")
+      endif()
     endif()
-    add_subdirectory(src)
-    add_subdirectory(include/sdf)
-    add_subdirectory(sdf)
-    add_subdirectory(conf)
-    add_subdirectory(doc)
-  endif(BUILD_SDF)
-
-  ########################################
-  # Setup Codecheck
-  include (IgnCodeCheck)
-  set(CPPCHECK_DIRS
-    ${PROJECT_SOURCE_DIR}/src
-    ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/test/integration
-    ${PROJECT_SOURCE_DIR}/test/performance)
-
-  set(CPPCHECK_INCLUDE_DIRS
-    ${PROJECT_BINARY_DIR}
-    ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/test/integration
-    ${PROJECT_SOURCE_DIR}/test/performance)
+  endif()
 
-  # Ignore vendored directories.
-  file(WRITE ${PROJECT_BINARY_DIR}/cppcheck.suppress
-    "*:${PROJECT_SOURCE_DIR}/src/urdf/*\n"
-    )
-  ign_setup_target_for_codecheck()
+  if (USE_INTERNAL_URDF)
+    message(STATUS "Using internal URDF")
+  endif()
 
-  ########################################
-  # Make the package config file
-  configure_file(${CMAKE_SOURCE_DIR}/cmake/sdformat_pc.in
-    ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME_LOWER}.pc @ONLY)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cmake/${PROJECT_NAME_LOWER}.pc DESTINATION
-    ${LIB_INSTALL_DIR}/pkgconfig COMPONENT pkgconfig)
+  #################################################
+  # Find ign command line utility:
+  ign_find_package(ignition-tools)
+
+  ################################################
+  # Find the Python interpreter for running the
+  # check_test_ran.py script
+  find_package(PythonInterp 3 QUIET)
+
+  ################################################
+  # Find psutil python package for memory tests
+  include (${PROJECT_SOURCE_DIR}/cmake/SDFUtils.cmake)
+  find_python_module(psutil)
+  if (NOT PY_PSUTIL)
+    ign_build_warning("Python psutil package not found. Memory leak tests will be skipped")
+  endif()
 
-  ########################################
-  # Configure documentation uploader
-  configure_file("${CMAKE_SOURCE_DIR}/cmake/upload_doc.sh.in"
-    ${CMAKE_BINARY_DIR}/upload_doc.sh @ONLY)
+  ################################################
+  # Find ruby executable to produce xml schemas
+  find_program(RUBY ruby)
```

Comment:
Would be a bit more elegant to use the `find_package(ruby)` + `RUBY_EXECUTABLE`, could cover some corner cases. Not a big deal anyway, leave it up to you.

Reply:
Actually, I think it might make things harder. I believe all we need is the ruby executable, but I think you can only `find_package(ruby)` if you have the development package with ruby headers installed. For our purposes, I think `find_program` gives us what we need and is easier to satisfy

Reply:
makes sense, thanks.